apps/wallet/src/ui/app/hooks/deepbook/useGetEstimate.ts

```diff
@@ -0,0 +1,287 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { useActiveAccount } from '_app/hooks/useActiveAccount';
+import { type WalletSigner } from '_app/WalletSigner';
+import {
+	DEEPBOOK_KEY,
+	DEFAULT_WALLET_FEE_ADDRESS,
+	ESTIMATED_GAS_FEES_PERCENTAGE,
+	ONE_SUI_DEEPBOOK,
+	WALLET_FEES_PERCENTAGE,
+} from '_pages/swap/constants';
+import { useDeepBookContext } from '_shared/deepBook/context';
+import { FEATURES } from '_src/shared/experimentation/features';
+import { useFeatureValue } from '@growthbook/growthbook-react';
+import { useGetObject } from '@mysten/core';
+import { useSuiClient } from '@mysten/dapp-kit';
+import { type DeepBookClient } from '@mysten/deepbook';
+import { TransactionBlock } from '@mysten/sui.js/builder';
+import { type CoinStruct, type SuiClient } from '@mysten/sui.js/client';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import BigNumber from 'bignumber.js';
+
+const MAX_COINS_PER_REQUEST = 10;
+
+async function getCoinsByBalance({
+	coinType,
+	balance,
+	suiClient,
+	address,
+}: {
+	coinType: string;
+	balance: string;
+	suiClient: SuiClient;
+	address: string;
+}) {
+	let cursor: string | undefined | null = null;
```

Comment:
can `cursor` be undefined?

Reply:
Yep

<img width="828" alt="Screenshot 2023-10-25 at 10 08 11 AM" src="https://github.com/MystenLabs/sui/assets/127577476/b26ecb68-5b57-4746-ad63-6d701712852c">